src/status_im/contexts/wallet/events/collectibles.cljs

```diff
@@ -0,0 +1,127 @@
+(ns status-im.contexts.wallet.events.collectibles
+  (:require [camel-snake-kebab.core :as csk]
+            [camel-snake-kebab.extras :as cske]
+            [clojure.string :as string]
+            [re-frame.core :as rf]
+            [taoensso.timbre :as log]
+            [utils.ethereum.chain :as chain]
+            [utils.transforms :as types]))
+
+(def collectibles-request-batch-size 1000)
+
+(defn displayable-collectible?
+  [collectible]
+  (let [{:keys [image-url animation-url]} (:collectible-data collectible)]
+    (or (not (string/blank? animation-url))
+        (not (string/blank? image-url)))))
+
+(defn- add-collectibles-to-accounts
+  [accounts collectibles]
+  (reduce (fn [acc {:keys [ownership] :as collectible}]
+            (->> ownership
+                 (map :address) ; In ERC1155 tokens a collectible can be owned by multiple addresses.
+                 (reduce (fn add-collectible-to-address [acc address]
+                           (update-in acc [address :collectibles] conj collectible))
+                         acc)))
+          accounts
+          collectibles))
+
+(defn store-collectibles
+  [{:keys [db]} [collectibles]]
+  (let [displayable-collectibles (filter displayable-collectible? collectibles)]
+    {:db (update-in db [:wallet :accounts] add-collectibles-to-accounts displayable-collectibles)}))
+
+(rf/reg-event-fx :wallet/store-collectibles store-collectibles)
+
+(defn clear-stored-collectibles
+  [{:keys [db]}]
+  {:db (update-in db [:wallet :accounts] update-vals #(dissoc % :collectibles))})
+
+(rf/reg-event-fx :wallet/clear-stored-collectibles clear-stored-collectibles)
+
+(defn store-last-collectible-details
+  [{:keys [db]} [collectible]]
+  {:db (assoc-in db [:wallet :last-collectible-details] collectible)})
+
+(rf/reg-event-fx :wallet/store-last-collectible-details store-last-collectible-details)
+
+(def collectible-data-types
+  {:unique-id        0
+   :header           1
+   :details          2
+   :community-header 3})
+
+(def fetch-type
+  {:never-fetch         0
+   :always-fetch        1
+   :fetch-if-not-cached 2
+   :fetch-if-cache-old  3})
+
+(def max-cache-age-seconds 3600)
```

Comment:
Maybe these can be moved to a constants file, or at the top of the file

Reply:
Will move it to the top :+1:

Reply:
done